lib/matplotlib/lines.py

```diff
@@ -1133,6 +1134,12 @@ def _split_drawstyle_linestyle(self, ls):
         """
         for ds in self.drawStyleKeys:  # long names are first in the list
             if ls.startswith(ds):
+                cbook.warn_deprecated(
+                    "3.1", message="Passing the drawstyle with the linestyle "
+                    "as a single string is deprecated since Matplotlib "
+                    "%(since)s and support will be removed %(removal)s; "
+                    "please pass the drawstyle separately ('drawstyle=...' or "
```

Comment:
-0.5 on quoting `'drawstyle=...'`. Might lead to confusion with strings. Alternative:

> please pass the drawstyle separately using the drawstyle (or ds) keyword arguments.

Reply:
done